src/System.Runtime/tests/System/Enum.cs

```diff
@@ -4,219 +4,165 @@
+    public static void TestGetName(object value, string expected)
+    {
+        String s = Enum.GetName(typeof(SimpleEnum), value);
+        Assert.Equal(expected, s);
     }
 
     [Fact]
-    public static void TestIsDefined()
+    public static void TestGetNameMultipleMatches()
     {
-        Type t = typeof(SimpleEnum);
-        bool b;
-
-        // Value can be a string...
-        b = Enum.IsDefined(t, "Red");
-        Assert.True(b);
-
-        b = Enum.IsDefined(t, "Green");
-        Assert.True(b);
-
-        b = Enum.IsDefined(t, "Blue");
-        Assert.True(b);
-
-        b = Enum.IsDefined(t, " Blue");
-        Assert.True(!b);
-
-        b = Enum.IsDefined(t, "blue");
-        Assert.True(!b);
-
-        // or an Enum value
-        b = Enum.IsDefined(t, SimpleEnum.Red);
-        Assert.True(b);
-
-        b = Enum.IsDefined(t, (SimpleEnum)(99));
-        Assert.True(!b);
-
-        // but not a different enum.
-        Assert.Throws<ArgumentException>(() => Enum.IsDefined(t, Int32Enum.One));
-
-        // or the underlying integer
-        b = Enum.IsDefined(t, 1);
-        Assert.True(b);
-
-        b = Enum.IsDefined(t, 99);
-        Assert.True(!b);
-
-        // but not just any integer type.
-        Assert.Throws<ArgumentException>(() => Enum.IsDefined(t, (short)1));
-
-        Assert.Throws<ArgumentException>(() => Enum.IsDefined(t, (uint)1));
+        // In the case of multiple matches, GetName returns one of them (which one is an implementation detail.)
+        String s = Enum.GetName(typeof(SimpleEnum), 3);
+        Assert.True(s == "Green" || s == "Green_a" || s == "Green_b");
+    }
 
-        // "Combos" do not pass.
-        b = Enum.IsDefined(typeof(Int32Enum), 0x1 | 0x2);
-        Assert.True(!b);
+    [Fact]
+    public static void TestGetNameInvalid()
+    {
+        Type t = typeof(SimpleEnum);
+        Assert.Throws<ArgumentNullException>("enumType", () => Enum.GetName(null, 1)); // Enum type is null
+        Assert.Throws<ArgumentNullException>("value", () => Enum.GetName(t, null)); // Value is null
+
+        Assert.Throws<ArgumentException>(null, () => Enum.GetName(typeof(Object), 1)); // Enum type is not an enum
+        Assert.Throws<ArgumentException>("value", () => Enum.GetName(t, "Red")); // Value is not the type of the enum's raw data
+        Assert.Throws<ArgumentException>(() => Enum.GetName(t, (IntPtr)0)); // Value is out of range
+    }
+
+    [Theory]
+    [InlineData(0xffffffffffffff80LU, "Min")]
+    [InlineData(0xffffff80u, null)]
+    [InlineData(unchecked((int)(0xffffff80u)), "Min")]
+    [InlineData(true, "One")]
+    [InlineData((char)1, "One")]
+    [InlineData(SimpleEnum.Red, "One")] // API doesnt't care if you pass in a completely different enum
+    public static void TestGetNameNonItegralTypes(object value, string expected)
+    {
+        /*
+            * Despite what MSDN says, GetName() does not require passing in the exact integral type.
```

Comment:
Does MSDN still say this?  If so, and it's incorrect, we should get the docs fixed.